pkgs/stdenv/generic/setup.sh

```diff
@@ -574,7 +578,11 @@ _addToEnv() {
                     ${pkgsHostTarget+"${pkgsHostTarget[@]}"} \
                     ${pkgsTargetTarget+"${pkgsTargetTarget[@]}"}
                 do
+                    if [[ "$visitedPkgs" = *"$pkg"* ]]; then
```

Comment:
Is this susceptible to false matches when package is a substring of another?

Oh, nevermind, these are probably store paths so that can't happen, right? :)